swagger.yaml

```diff
@@ -0,0 +1,211 @@
+    post:
+      tags:
+      - "Code List"
+      summary: "Get a set of code lists containing information about dimensions"
+      description: "Get a list of codes containing information about dimensions which are used for all datasets at the ONS"
+      parameters:
+      - $ref: '#/parameters/pageSize'
+      - $ref: '#/parameters/pageNumber'
+      - $ref: '#/parameters/query'
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "A Json message containing a set of code lists"
+          schema:
+            $ref: '#/definitions/CodeLists'
+        400:
+          description: "Missing parameters within request or invalid query"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /code-list/{id}:
+    get:
+      tags:
+      - "Code List"
+      summary: "Get a code list"
+      description: "Get information about a code list"
+      parameters:
+      - $ref: '#/parameters/id'
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Json object containing information about the code list"
+          schema:
+            $ref: '#/definitions/CodeList'
+        404:
+          description: "Code list not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /code-list/{id}/codes:
+    get:
+      tags:
+       - "Code List"
+      summary: "Get a list of all codes within a code list type"
+      description: "Get a list of codes within a code list type"
+      parameters:
+      - $ref: '#/parameters/id'
+      - $ref: '#/parameters/pageSize'
+      - $ref: '#/parameters/pageNumber'
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "A Json message containing a list of Codes"
+          schema:
+            $ref: '#/definitions/Codes'
+        404:
+          description: "Code list not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /code-list/{id}/codes/{codeId}:
+    get:
+      tags:
+       - "Code List"
+      summary: "Get in depth information about a code"
+      description: "Get information about a code within a code list"
+      parameters:
+      - $ref: '#/parameters/id'
+      - $ref: '#/parameters/codeId'
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Get in depth information about a code"
+          schema:
+            $ref: '#/definitions/CodeInformation'
+        404:
+          description: "Code list or code not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+
+definitions:
+  CodeList:
+    type: object
+    properties:
+      id:
+        type: string
+        description: "The id for the code list"
+      name:
+        type: string
+        description: "The name of the code list"
+      metadata:
```

Comment:
At the moment, I think we only know that the metadata looks like for a dataset. I think there been no talks about how the metadata will be like for dimensions. This is why there no model present.